### PR TITLE
DT-727 Spring Dependency Manager and Java functional test

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/DpsSpringBootPluginTest.kt [hunk 1]
```diff
@@ -13,64 +19,90 @@ import org.springframework.boot.gradle.plugin.SpringBootPlugin
...
+  }
+
+  @Nested
+  inner class Plugins {
 
-    @Nested
-    inner class Plugins {
+    @Test
+    fun `Should apply the Spring Boot plugin`() {
+      val thrown = catchThrowable { project.plugins.getPlugin(SpringBootPlugin::class.java) }
```
Comment (petergphillips) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/DpsSpringBootPluginTest.kt:34]: why not just let the exception propagate instead i.e. 
```
project.plugins.getPlugin(SpringBootPlugin::class.java)
```

Comment (petergphillips) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/DpsSpringBootPluginTest.kt:34]: the exception will then be thrown by the test method and will fail

Comment (mikehalmamoj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/DpsSpringBootPluginTest.kt:34]: Yeah I thought about that, but it's not always obvious that a test without an assertion is actually testing the lack of an exception. Whereas the above is obvious.  But... :man_shrugging: I'll switch back to your way in the next PR.

Comment (petergphillips) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/DpsSpringBootPluginTest.kt:34]: might therefore add a comment above along the lines of
```
// will throw exception if plugin fails to apply
```

**Pull request merged**